pkg/networkservice/mechanisms/directmemif/metrics/collector.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package metrics - contains directmemif metrics and metrics collector
+package metrics
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// RxBytes is a total number of bytes received from target
+	RxBytes = "rx_bytes"
+	// TxBytes is a total number of bytes transmitted to target
+	TxBytes = "tx_bytes"
+)
+
+// Collector aggregates metrics
+type Collector interface {
+	Update(map[string]string) error
+	Metrics() map[string]string
+}
+
+// Collector aggregates direct memif metrics
+type directMemifCollector struct {
```

Comment:
I do not think that we actually need to create a separate entity for collecting metrics in directmemif. 

I would suggest to just add new method for proxy
```
type Proxy interface {
   Start() error
   End() error
   Metrics() map[string]string
}
```

And just use this in directmemif server

```
path.GetPathSegments()[path.GetIndex()].Metrics = d.proxies[connectionID].Metrics()
```

Reply:
Collector interface removed. Added `Metrics()` method to proxy.